llvm/tools/llvm-mca/Views/InstructionInfoView.cpp

```diff
@@ -29,82 +60,151 @@ void InstructionInfoView::printView(raw_ostream &OS) const {
   IIVDVec IIVD(Source.size());
   collectData(IIVD);
 
-  TempStream << "\n\nInstruction Info:\n";
-  TempStream << "[1]: #uOps\n[2]: Latency\n[3]: RThroughput\n"
-             << "[4]: MayLoad\n[5]: MayStore\n[6]: HasSideEffects (U)\n";
+  if (PrintFullInfo) {
+    FOS << "\n\nResources:\n";
+    const MCSchedModel &SM = getSubTargetInfo().getSchedModel();
+    for (unsigned I = 1, ResourceIndex = 0, E = SM.getNumProcResourceKinds();
+         I < E; ++I) {
+      const MCProcResourceDesc &ProcResource = *SM.getProcResource(I);
+      unsigned NumUnits = ProcResource.NumUnits;
+      // Skip invalid resources with zero units.
+      if (!NumUnits)
+        continue;
+
+      FOS << '[' << ResourceIndex << ']';
+      FOS.PadToColumn(6);
+      FOS << "- " << ProcResource.Name << ':' << NumUnits;
+      if (ProcResource.SubUnitsIdxBegin) {
+        FOS.PadToColumn(20);
+        for (unsigned U = 0; U < NumUnits; ++U) {
+          FOS << SM.getProcResource(ProcResource.SubUnitsIdxBegin[U])->Name;
+          if ((U + 1) < NumUnits)
+            FOS << ", ";
+        }
+      }
+      FOS << '\n';
+      ResourceIndex++;
+    }
+  }
+
+  std::vector<unsigned> Paddings = {0, 7, 14, 21, 28, 35};
+  std::vector<std::string> Fields = {"#uOps",       "Latency",
+                                     "RThroughput", "MayLoad",
+                                     "MayStore",    "HasSideEffects (U)"};
+  std::vector<std::string> EndFields;
+  unsigned LastPadding = 35;
+  if (PrintFullInfo) {
+    Fields.push_back("Bypass Latency");
+    Paddings.push_back(LastPadding + 7);
+    LastPadding += 7;
```

Comment:
@topperc spot this yesterday:
```suggestion
    LastPadding += 7;
    Paddings.push_back(LastPadding);
```
ditto every other similar cases in this file